File: app/src/main/java/com/kelsos/mbrc/connection_manager/ConnectionManagerModule.java
package com.kelsos.mbrc.connection_manager;


import android.support.annotation.NonNull;

import toothpick.config.Module;

public class ConnectionManagerModule extends Module{
  private ConnectionManagerModule() {
    bind(ConnectionManagerPresenter.class).to(ConnectionManagerPresenterImpl.class).scope();
  }

  @NonNull public static ConnectionManagerModule create() {
    return new ConnectionManagerModule();
  }
}

File: app/src/main/kotlin/com/kelsos/mbrc/connection_manager/ConnectionManagerModule.kt
package com.kelsos.mbrc.connection_manager


import toothpick.config.Module

class ConnectionManagerModule private constructor() : Module() {
  init {
    bind<ConnectionManagerPresenter>(ConnectionManagerPresenter::class.java).to(ConnectionManagerPresenterImpl::class.java).singletonInScope()
  }

  companion object {

    fun create(): ConnectionManagerModule {
      return ConnectionManagerModule()
    }
  }
}
